app/src/main/java/at/roteskreuz/stopcorona/model/managers/DatabaseCleanupManager.kt

```diff
@@ -72,23 +70,55 @@ class DatabaseCleanupManagerImpl(
         launch {
             val configuration = configurationRepository.observeConfiguration().blockingFirst()
 
-            val thresholdRedMessages = ZonedDateTime.now().minusHours(configuration.redWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.InfectionLevel.Red, thresholdRedMessages)
-
-            val thresholdYellowMessages = ZonedDateTime.now().minusHours(configuration.yellowWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.InfectionLevel.Yellow, thresholdYellowMessages)
-
-            val thresholdGreenMessages = ZonedDateTime.now().minusDays(THRESHOLD_REMOVE_INCOMING_GREEN_MESSAGES)
-            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(MessageType.Revoke.Suspicion, thresholdGreenMessages)
+            val thresholdRedMessages = ZonedDateTime.now()
+                .minusHours(configuration.redWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
+            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(
+                MessageType.InfectionLevel.Red,
+                thresholdRedMessages
+            )
+
+            val thresholdYellowMessages = ZonedDateTime.now()
+                .minusHours(configuration.yellowWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
+            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(
+                MessageType.InfectionLevel.Yellow,
+                thresholdYellowMessages
+            )
+
+            val thresholdGreenMessages =
+                ZonedDateTime.now().minusDays(THRESHOLD_REMOVE_INCOMING_GREEN_MESSAGES)
+            infectionMessageDao.removeReceivedInfectionMessagesOlderThan(
+                MessageType.Revoke.Suspicion,
+                thresholdGreenMessages
+            )
         }
     }
 
     private fun cleanupSentTemporaryExposureKeys() {
         launch {
-            // The sent temporary exposure keys are removed in case we detect that we are not anymore in quarantine.
-            if (quarantineRepository.getQuarantineStatus() is QuarantineStatus.Free) {
-                temporaryExposureKeysDao.removeSentTemporaryExposureKeys()
-            }
+            val configuration = configurationRepository.observeConfiguration().blockingFirst()
+
+            val nowAsRollingStartIntervalNumber = ZonedDateTime.now()
+                .toRollingStartIntervalNumber()
+            temporaryExposureKeysDao.removeSentInfectionMessagesOlderThan(
+                MessageType.Revoke.Suspicion,
+                nowAsRollingStartIntervalNumber
+            )
+
+            val thresholdYellowMessageEndOfTheDayAsRollingStart = ZonedDateTime.now()
+                .minusHours(configuration.yellowWarningQuarantine?.toLong() ?: Long.MAX_VALUE)
```

Comment:
Long.MAX_VALUE move to configuration

Reply:
Long.MAX_VALUE is not a general default for `configuration.yellowWarningQuarantine` in some cases Long.MIN_VALUE could be suitable.

MAX_VALUE is the value that makes sense in this situation when yellowWarningQuarantine is null.

Reply:
As discussed, use here 1. 1. 1970 as the threshold in case of configuration value is null. Meaning, we won't delete anything.